src/matrices.jl

```diff
@@ -0,0 +1,108 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    JMatrix{T}
+
+Square matrix of ones.
+"""
+struct JMatrix{T} end
+(J::JMatrix{T})(d::Integer) where {T} = Ones{T}(d, d)
+*(::JMatrix, v::AbstractVector) = Fill(sum(v), length(v))
+*(v::Adjoint{<:Any, <:AbstractVector}, J::JMatrix) = (J*v')'
+
+"""
+    FMatrix{T}
+
+`F` matrix, as defined by Aitchison 1986.
+"""
+struct FMatrix{T} end
+(F::FMatrix{T})(d::Integer) where {T} = [Diagonal(fill(one(T), d)) -Ones{T}(d)]
+*(::FMatrix{T}, v::AbstractVector) where {T} = Fill(-last(v), length(v)-1) + v[1:end-1]
+*(v::Adjoint{<:Any, <:AbstractVector}, ::FMatrix) = [v -sum(v)]
+
+"""
+    GMatrix{T}(N)
+
+`G` matrix, as defined by Aitchison 1986.
+"""
+struct GMatrix{T} end
+(G::GMatrix{T})(D::Integer) where {T} = -ones(T, D, D)/D + Diagonal(fill(one(T), D))
+*(::GMatrix, v::AbstractVector) = Fill(-sum(v)/length(v), length(v)) + v
+*(v::Adjoint{<:Any, <:AbstractVector}, G::GMatrix) = (G*v')'
+
+"""
+    HMatrix{T}(d)
+
+`H` matrix, as defined by Aitchison 1986.
+"""
+struct HMatrix{T} end
+(H::HMatrix{T})(d::Integer) where {T} = ones(T, d, d) + Diagonal(fill(one(T), d))
+*(::HMatrix, v::AbstractVector) = Fill(sum(v), length(v)) + v
+*(v::Adjoint{<:Any, <:AbstractVector}, H::HMatrix) = (H*v')'
+
+"""
+    J
+    J(d)
+
+User interface for JMatrix, the matrix of ones.
+
+## Examples
+
+```
+julia> J(3)
+julia> J*v
+julia> v'*J
+```
+"""
+const J = JMatrix{Bool}()
+
+"""
+    F
+    F(d)
+
+User interface for FMatrix, as defined by Aitchison.
```

Comment:
Try to explain what are the entries F[i,j] of the matrix definition in terms of d and D.

Reply:
And this comment applies to the other matrix definitions as well. The docstring should be self-contained so that anyone can grasp what is happening quickly.

Reply:
done